app/bundles/UserBundle/Model/PasswordStrengthEstimatorModel.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\UserBundle\Model;
+
+use ZxcvbnPhp\Zxcvbn as PasswordStrengthEstimator;
+
+class PasswordStrengthEstimatorModel
+{
+    public const MINIMUM_PASSWORD_STRENGTH_ALLOWED = 3;
+
+    private const DICTIONARY = [
+        'mautic',
+        'user',
+        'lead',
+        'bundle',
+        'campaign',
+        'company',
+    ];
+
+    private PasswordStrengthEstimator $passwordStrengthEstimator;
+
+    public function __construct()
+    {
+        $this->passwordStrengthEstimator = new PasswordStrengthEstimator();
+    }
+
+    /**
+     * @param string[] $dictionary
+     */
+    public function validate(?string $password, int $score = self::MINIMUM_PASSWORD_STRENGTH_ALLOWED, array $dictionary = self::DICTIONARY): bool
+    {
+        return $score <= $this->passwordStrengthEstimator->passwordStrength($password, $this->sanitazeDictionary($dictionary))['score'];
```

Comment:
Could we emit an event here to add the ability to set custom password restrictions via plugins?

Reply:
Isn't `UserEvents::USER_FORM_POST_LOCAL_PASSWORD_AUTHENTICATION` the event you could hook into?

Reply:
It's only triggered on login. I couldn't add requirements for a new password

Reply:
I see. Feel free to dispatch the event. If you cannot add it to this PR directly, create a new one from this branch and send it to me. I can cherry-pick it to this PR.

Reply:
I've made a PR to your fork with these changes